### PR TITLE
completed pagination for both briefs and freelancer search pages

### PR DESCRIPTION
**Pull request by vadik0101**

### CODE REVIEW

File: src/pages/briefs/index.tsx [hunk 1]
```diff
@@ -449,6 +478,40 @@ const Briefs = (): JSX.Element => {
             </div>
           ))}
         </div>
+        <Pagination
+          pageSize={3}
```
Comment (cuteolaf) [inline src/pages/briefs/index.tsx:482]: looks too small
should be able to select number of items to show on a page.

File: src/pages/briefs/index.tsx [hunk 2]
```diff
@@ -338,6 +340,33 @@ const Briefs = (): JSX.Element => {
     setFilterVisible(!filterVisble);
   };
 
+  const paginatedBriefs = (): BriefStepProps => {
+    const briefsPerPage = 3;
```
Comment (cuteolaf) [inline src/pages/briefs/index.tsx:344]: this should be a param

File: src/pages/briefs/index.tsx [hunk 3]
```diff
@@ -449,6 +478,40 @@ const Briefs = (): JSX.Element => {
             </div>
           ))}
         </div>
+        <Pagination
+          pageSize={3}
+          total={briefs.length}
+          onChange={(page: number, pageSize: number) =>
+            paginatedBriefs()?.paginate(page)
```
Comment (cuteolaf) [inline src/pages/briefs/index.tsx:485]: this should be updated based on `paginatedBriefs`

like `paginatedBriefs(pageSize)?.paginate(page)`

File: src/pages/briefs/index.tsx [hunk 4]
```diff
@@ -449,6 +478,40 @@ const Briefs = (): JSX.Element => {
...
+          jumpNextIcon={
+            <div className="h-[32px] hover:bg-[--theme-primary]  hover:text-black mr-6  rounded-[4px] border border-primary w-[32px] pt-1 items-center text-center text-sm !font-bold text-primary ">
+              {">>"}
+            </div>
+          }
+          jumpPrevIcon={
+            <div className="h-[32px] hover:bg-[--theme-primary]  hover:text-black  mr-6  rounded-[4px] border border-primary w-[32px] pt-1 items-center text-center text-sm !font-bold text-primary">
+              {"<<"}
+            </div>
+          }
```
Comment (cuteolaf) [inline src/pages/briefs/index.tsx:513]: DRY
same style used for arrows

File: src/pages/freelancers/index.tsx [hunk 5]
```diff
@@ -277,7 +274,37 @@ const Freelancers = (): JSX.Element => {
     setFilterVisible(!filterVisble);
   };
 
-  if(loading) return <LoadingFreelancers/>
+  const paginatedFreelancers = (): FreelancerStepProps => {
+    const freelancersPerPage = 3;
```
Comment (cuteolaf) [inline src/pages/freelancers/index.tsx:278]: same here

File: src/pages/freelancers/index.tsx [hunk 6]
```diff
@@ -353,40 +380,83 @@ const Freelancers = (): JSX.Element => {
...
-                    <h5>{title}</h5>
-                    <div className={styles.skills}>
-                      {skills?.slice(0, 3).map((skill, index) => (
-                        <p className={styles.skill} key={index}>
-                          {skill.name}
-                        </p>
-                      ))}
+            {freelancers?.length &&
+              paginatedFreelancers?.()
+                ?.currentFreelancers?.slice?.(0, 10)
```
Comment (cuteolaf) [inline src/pages/freelancers/index.tsx:385]: ?

File: src/pages/freelancers/index.tsx [hunk 7]
```diff
@@ -353,40 +380,83 @@ const Freelancers = (): JSX.Element => {
...
-                  >
-                    View
-                  </button>
-                </div>
-              ))}
+                  )
+                )}
           </div>
+          <Pagination
+            pageSize={3}
```
Comment (cuteolaf) [inline src/pages/freelancers/index.tsx:427]: `pageSize` should be configurable.

File: src/pages/freelancers/index.tsx [hunk 8]
```diff
@@ -353,40 +380,83 @@ const Freelancers = (): JSX.Element => {
...
+            total={freelancers.length}
+            onChange={(page: number, pageSize: number) =>
+              paginatedFreelancers()?.paginate(page)
```
Comment (cuteolaf) [inline src/pages/freelancers/index.tsx:430]: same issue as the brief search page

File: src/pages/freelancers/index.tsx [hunk 9]
```diff
@@ -353,40 +380,83 @@ const Freelancers = (): JSX.Element => {
...
+            }
+            className="flex flex-row items-center my-10 px-10"
+            itemRender={(page, type, originalElement) => {
+              if (type === "page") {
+                return <PageItem page={page} />;
+              }
+              return originalElement;
+            }}
+            prevIcon={
+              <div className="h-[32px] hover:bg-[--theme-primary] hover:text-black mr-6 cursor-pointer rounded-[4px] border border-primary w-[32px] pt-1 items-center text-center text-sm !font-bold text-primary">
```
Comment (cuteolaf) [inline src/pages/freelancers/index.tsx:440]: please don't repeat yourself

**Review by cuteolaf — CHANGES_REQUESTED** [review]

File: src/pages/briefs/index.tsx (hunk 1 above)
Comment (vadik0101) [reply src/pages/briefs/index.tsx:482]: the items are large and 3 is more than enough for a single page scroll

File: src/pages/freelancers/index.tsx (hunk 6 above)
Comment (vadik0101) [reply src/pages/freelancers/index.tsx:385]: optional chaining

File: src/pages/freelancers/index.tsx (hunk 9 above)
Comment (vadik0101) [reply src/pages/freelancers/index.tsx]: fixed

File: src/pages/freelancers/index.tsx (hunk 8 above)
Comment (vadik0101) [reply src/pages/freelancers/index.tsx:432]: made the page a static property for now , pending when the UI is updated

File: src/pages/freelancers/index.tsx (hunk 5 above)
Comment (vadik0101) [reply src/pages/freelancers/index.tsx]: updated

File: src/pages/briefs/index.tsx (hunk 4 above)
Comment (vadik0101) [reply src/pages/briefs/index.tsx]: thanks, updated

File: src/pages/briefs/index.tsx (hunk 3 above)
Comment (vadik0101) [reply src/pages/briefs/index.tsx:488]: made it a variable that can be updated

File: src/pages/briefs/index.tsx (hunk 2 above)
Comment (vadik0101) [reply src/pages/briefs/index.tsx]: made it a variable that can be updated

File: src/pages/briefs/index.tsx (hunk 1 above)
Comment (vadik0101) [reply src/pages/briefs/index.tsx]: sure

**Review by samelamin — APPROVED** [review]

**Pull request merged**